tools/autograd/derivatives.yaml

```diff
@@ -367,6 +367,9 @@
 - name: exp(Tensor self) -> Tensor
   self: grad * result
 
+- name: exp2(Tensor self) -> Tensor
+  self: grad * result * 0.6931471805599453
```

Comment:
@albanD Are you happy with this approach or would you prefer the approach taken in, for example, the deg2rad PR (https://github.com/pytorch/pytorch/pull/38852/files)? 

@ShawnZhong, did you chose this version of ln(2) because it goes to the first decimal s.t. the number is no longer exactly representable as a double?

Reply:
I would be slightly concerned about the precision here for the double case. Is that done on purpose?
If we want full precision, adding a constexpr sounds like a good way to easily define the full precision value.

Reply:
IEEE 1003.1 says that math.h (AKA cmath) should define M_LN2, which we should try to use instead of our own macro/constexpr expression.